controllers/storagecluster/storageconsumer.go

```diff
@@ -57,39 +60,47 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 		return ctrl.Result{}, fmt.Errorf("failed to generate volumegroupsnapshotclasses list for distribution: %v", err)
 	}
 
-	storageConsumer := &ocsv1a1.StorageConsumer{}
-	storageConsumer.Name = defaults.LocalStorageConsumerName
-	storageConsumer.Namespace = storageCluster.Namespace
-	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageConsumer, func() error {
-		if err := controllerutil.SetControllerReference(storageCluster, storageConsumer, r.Scheme); err != nil {
+	consumerName, err := getInternalStorageConsumerName(r, storageCluster.Name, storageCluster.Namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get internal storage consumer name: %v", err)
+	}
+
+	storageConsumerObj := &ocsv1a1.StorageConsumer{}
```

Comment:
why the rename? I don't see a reason to call this out which also introduces not mandatory diff.

Reply:
It was creating confusion as we have a struct with the same name, I could preserve the original name

Reply:
ouch, you are fixing an actual bug of overshadowing type. fine.